pkg/operator/certmonitor/monitor.go

```diff
@@ -0,0 +1,196 @@
+package certmonitor
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/klog"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/openshift/library-go/pkg/crypto"
+)
+
+const (
+	// CertificateManagedLabelName marks config map or secret as object that contains managed certificates.
+	// This groups all objects that store certs and allow easy query to get them all.
+	// Also any object with this label will be collected by prometheus and transformed to metrics.
+	// The value of this label should be set to "true".
+	CertificateManagedLabelName = "auth.openshift.io/certificate-managed"
+
+	// CertificateType indicates what kind of certificate the object contain. Can be of of "signer", "target" or "ca-bundle".
+	CertificateTypeAnnotationName = "auth.openshift.io/certificate-type"
```

Comment:
seems like the presence of this implies the one above

Reply:
True, but you can't have selectors over labels.

Reply:
can you make label selector match everything just having the label regardless of its value?

Reply:
Yeah, I think so.  have a look a the syntax

Reply:
yeah, labels.Parse() and `cert-type in (foo, bar)`